docs/source/tutorials/fashion_mnist.ipynb

```diff
@@ -0,0 +1,1274 @@
+    "            outputs = net(images)\n",
+    "            # the label with the highest energy will be our prediction\n",
+    "            _, predicted = torch.max(outputs.data, 1)\n",
+    "            total += labels.size(0)\n",
+    "            accuracy += (predicted == labels).sum().item()\n",
+    "\n",
+    "    # compute the accuracy over all test images\n",
+    "    accuracy = 100 * accuracy / total\n",
+    "    return accuracy\n",
+    "\n",
+    "\n",
+    "def train(trainloader, testloader, n_epochs, patience):\n",
+    "    model = Net()\n",
+    "    criterion = nn.CrossEntropyLoss()\n",
+    "    optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.9)\n",
+    "    model = model.to(device)\n",
+    "\n",
+    "    best_test_accuracy = 0.0\n",
+    "    for epoch in range(n_epochs):  # loop over the dataset multiple times\n",
+    "        running_loss = 0.0\n",
+    "\n",
+    "        for _, data in enumerate(trainloader, 0):\n",
+    "            # get the inputs; data is a list of [inputs, labels]\n",
+    "            inputs, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            # zero the parameter gradients\n",
+    "            optimizer.zero_grad()\n",
+    "\n",
+    "            # forward + backward + optimize\n",
+    "            outputs = model(inputs)\n",
+    "            loss = criterion(outputs, labels)\n",
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "            # print statistics\n",
+    "            running_loss += loss.item()\n",
+    "\n",
+    "        accuracy = get_test_accuracy(model, testloader)\n",
+    "        print(\n",
+    "            f\"epoch: {epoch + 1} loss: {running_loss / len(trainloader):.3f} test acc: {accuracy}\"\n",
+    "        )\n",
+    "\n",
+    "        running_loss = 0.0\n",
+    "\n",
+    "        if accuracy > best_test_accuracy:\n",
+    "            best_epoch = epoch\n",
+    "\n",
+    "        if epoch - best_epoch > patience:\n",
+    "            print(f\"Early stopping at epoch {epoch + 1}\")\n",
+    "            break\n",
+    "    return model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Create K folds of the dataset for cross validation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "K = 3\n",
+    "\n",
+    "kfold = KFold(n_splits=K, shuffle=True, random_state=SEED)\n",
+    "splits = kfold.split(transformed_dataset)\n",
+    "\n",
+    "train_id_list, test_id_list = [], []\n",
+    "\n",
+    "for fold, (train_ids, test_ids) in enumerate(splits):\n",
+    "    train_id_list.append(train_ids)\n",
+    "    test_id_list.append(test_ids)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Train a model for each fold"
```

Comment:
this code structure is weird, why don't you just make the predictions for each validation set right here inside the loop. Note sure why you are running a separate loop later to get the predictions for each validation set.